### PR TITLE
Adding a reports download redirect page

### LINKED ISSUE #3280: UI: Download redirect page for scheduled reports
The scheduled reports reconciler itself does not generate an export. Instead, to support lazy evaluation of reports, it generates a "baked query" that should be passed to the existing `rill.runtime.v1.RuntimeService/Export` RPC to obtain a download link.

We need a UI page that reads the baked query from the URL args, resolves runtime credentials, and runs the normal export flow (call `Export` and trigger the download link that is returned).

Here is the code in the reports reconciler that builds the UI redirect link: https://github.com/rilldata/rill/blob/1db92f285294b649b48f6cb1191297e5acbeb9c5/runtime/reconcilers/report.go#L272-L278

Task list:

- [ ] Add the redirect page on `ui.rilldata.com/org/project/-/reports/<report id>/download?format=CSV&query=<BAKED_TOKEN>` (report ID does not matter for this purpose, but will be useful to easily navigate to report metadata after download)
- [ ] The page should resolve runtime host and JWT (and show appropriate errors if the user doesn't have access to the project)
- [ ] Then call the usual export flow on the runtime, passing `format` and `baked_query` from the URL params to the `Export` RPC

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #3280

## Summary by CodeRabbit

New Feature:
- Added a new functionality to download reports directly from the project interface. This feature enhances the user experience by providing a straightforward way to access and download reports.

Improvement:
- Implemented error handling for the report download process. If certain conditions are not met, an error message will be displayed to the user, improving the overall usability and reliability of the application.

### CODE REVIEW

File: web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  import type { V1ExportFormat } from "@rilldata/web-common/runtime-client";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+
+  $: reportId = $page.params.report;
+  $: format = $page.url.searchParams.get("format");
+  $: bakedQuery = $page.url.searchParams.get("query");
+  $: limit = $page.url.searchParams.get("limit");
+
+  const downloadReportMutation = createDownloadReportMutation();
+  let once = false;
```
Comment (ericpgreen2) [inline web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte:18]: Agree with CodeRabbit! `mutationRunOnce` would be more clear

File: web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  let error: string;
+  $: if (!format) {
+    error = "format is required";
+  } else if (!bakedQuery) {
+    error = "query is required";
+  } else if ($downloadReportMutation.error) {
+    error =
+      $downloadReportMutation.error.response?.data?.message ?? "unknown error";
+  }
```
[the remark this answers is not in the document]
Comment (ericpgreen2) [reply web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte:43]: Agree with the proposed commit

**Review by ericpgreen2 — APPROVED** [review]
Looks good 👍 . Approving so you can merge after addressing the comments.

**Comment by ericpgreen2** [comment]
Also, the page should be moved to `{org}/{project}/-/reports/{report}/export`.

(Add the hyphen between project and reports, and "download" -> "export")

File: web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte (hunk 1 above)
Comment (AdityaHegde) [reply web-admin/src/routes/[organization]/[project]/reports/[report]/download/+page.svelte:18]: Sure. I am naming it `downloadOnce` instead. `mutationRunOnce` is a little too generic.

**Pull request merged**